### PR TITLE
test(ticdc): skip kafka test for integration test clustered_index temporarily

### LINKED ISSUE #4727: integration test-clustered_index fail
### Which jobs are flaking?

https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_kafka_integration_test/detail/cdc_ghpr_kafka_integration_test/9308/pipeline

### Which test(s) are flaking?

integration test-clustered_index

### Jenkins logs or GitHub Actions link

```log
(paste logs here)
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #4727 

### What is changed and how it works?
Remove integration test clustered_index temporarily since kafka test fail

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: tests/integration_tests/clustered_index/run.sh [hunk 1]
```diff
@@ -44,7 +44,10 @@ function run() {
 	cleanup_process $CDC_BINARY
 }
 
-trap stop_tidb_cluster EXIT
-run $*
-check_logs $WORK_DIR
-echo "[$(date)] <<<<<< run test case $TEST_NAME success! >>>>>>"
+# kafka is not supported yet.
+if [ "$SINK_TYPE" != "kafka" ]; then
```
Comment (hi-rustin) [inline tests/integration_tests/clustered_index/run.sh:48]: Typo?

Please return directly and link the issue in the comment.

File: tests/integration_tests/clustered_index/run.sh [hunk 2]
```diff
@@ -45,9 +45,12 @@ function run() {
...
-	trap stop_tidb_cluster EXIT
-	run $*
-	check_logs $WORK_DIR
-	echo "[$(date)] <<<<<< run test case $TEST_NAME success! >>>>>>"
+# ref to issue: https://github.com/pingcap/tiflow/issues/3421
+if [ "$SINK_TYPE" = "kafka" ]; then
+	echo "[$(date)] <<<<<< skip test case $TEST_NAME for kafka! >>>>>>"
+	exit 0
 fi
+trap stop_tidb_cluster EXIT
```
Comment (hi-rustin) [inline tests/integration_tests/clustered_index/run.sh:53]: If we skip before running you don't need this logic?

Comment (maxshuang) [reply tests/integration_tests/clustered_index/run.sh:53]: I have not started the cluster, why I still  need to exit the cluster?

Comment (hi-rustin) [reply tests/integration_tests/clustered_index/run.sh:53]: Sorry, I seem to be running into a GitHub bug. It shows that you added the new code.

**Review by hi-rustin — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**